includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -901,20 +903,35 @@ public function process_refund( $order_id, $amount = null, $reason = '' ) {
 			return $response;
 
 		} elseif ( ! empty( $response->id ) ) {
-			$order->update_meta_data( '_stripe_refund_id', $response->id );
-
-			$amount = wc_price( $response->amount / 100 );
-
+			$formatted_amount = wc_price( $response->amount / 100 );
 			if ( in_array( strtolower( $order->get_currency() ), WC_Stripe_Helper::no_decimal_currencies() ) ) {
-				$amount = wc_price( $response->amount );
+				$formatted_amount = wc_price( $response->amount );
+			}
+
+			// If charge wasn't captured, skip creating a refund and cancel order.
+			if ( 'yes' !== $captured ) {
+				/* translators: dollar amount */
+				$order->add_order_note( sprintf( __( 'Pre-Authorization for %s voided.', 'woocommerce-gateway-stripe' ), $formatted_amount ) );
+				$order->update_status( 'cancelled' );
+				// If amount is set, that means this function was called from the manual refund form.
+				if ( ! is_null( $amount ) ) {
+					// Throw an exception to provide a custom message on why the refund failed.
+					throw new Exception( __( 'The authorization was voided and the order cancelled. Click okay to continue, then refresh the page.', 'woocommerce-gateway-stripe' ) );
+				}
+				else {
```

Comment:
This should be in the same line, but we can wait until we merge #1445 and fix it then

Reply:
😀  Will change it.